### ISSUE TITLE
Substring request path matching does not work with multiple context in 0.2.2 [PAXWEB-59]

### ISSUE THREAD

**Issue by ops4j-issues** [ops4j/org.ops4j.pax.web#372]
**[Alin Dreghiciu](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:a786db11-2013-41fb-af1d-ca5eb8f74aa0)** created **[PAXWEB-59](https://ops4j1.jira.com/browse/PAXWEB-59)** 

After the separation of servlet contexts per http contexts (in 0.2.2) the following situation could appear:

* in context 1 two aliases were registered "/foo" and "/"
* in context 2 one alias was registered "/bar"<br /> The registrations were done in this order.

Now, if you have a request for "bar/any" that should be handled in the context 2 by registration for "/bar", pax web was using the "/" registration from the first context.

Another wrong selected registration was possible if you had the following:

* in context 1 two aliases were registered "/foo"
* in context 2 one alias was registered "/foo/bar"<br /> In case of a request for "/foo/bar" or "/foo/bar/any" that should be handled by registration from context 2, pax web was selecting the first registration (for "/foo".

So, basically if there is more then one context and there are registered aliases that would match the request the first one to match a substring was selected without looking for a better match.

---

**Affects:** 0.2.2
**Fixed in:** 0.2.3
Votes: 0, Watches: 0
